### PR TITLE
feat(eks): timeout option helm charts

### LINKED ISSUE #8215: eks: allow specifying timeouts for HelmCharts
For long running helm charts the creation times out. 

### Reproduction Steps

I had this issue with the following resource (based on a minimal cluster defined through the CDK Cluster component).

    new eks.HelmChart(this, 'spinnaker', {
      cluster,
      chart: 'spinnaker',
      repository: 'https://kubernetes-charts.storage.googleapis.com/',
      version: '2.0.0-rc3',
      namespace: 'spinnaker'
    });

### Error Log

3/9 | 3:05:54 PM | CREATE_FAILED        | Custom::AWSCDK-EKS-HelmChart          | spinnaker/Resource/Default (spinnakerC4B2DB2C) Failed to create resource. TimeoutError: Connection timed out after 120000ms

### Environment

  - **CLI Version      :1.41.0 (build 9e071d2)
  - **Framework Version: 1.41.0
  - **OS               : macOS 10.14.6
  - **Language         : typescript

### Other

The helm command has an option for timeout, and in fact the chart I was trying to run (stable/spinnaker) suggests a timeout of 600s: https://github.com/helm/charts/blob/master/stable/spinnaker/README.md. I believe that the chart should add timeout as a option:

https://github.com/aws/aws-cdk/blob/86eac6af074bf78a921c52d613eca0dd4a514a49/packages/%40aws-cdk/aws-eks/lib/helm-chart.ts#L8-L50
1.41.0 (build 9e071d2)

--- 

This is :bug: Bug Report

### PR DESCRIPTION
**Pull request by eduardomourar**
This creates an additional option called `timeout` that will be passed down whenever deploying helm chart to an EKS cluster.

In order to allow the timeout parameter to work while performing helm commands, the provider framework has to honor the maximum timeout of 15 minutes from target process (lambda in this case).

closes #8215 

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: packages/@aws-cdk/custom-resources/lib/provider-framework/runtime/outbound.ts [hunk 1]
```diff
@@ -24,6 +25,8 @@ async function defaultStartExecution(req: AWS.StepFunctions.StartExecutionInput)
     sfn = new AWS.StepFunctions();
   }
 
+  sfn.config.update({httpOptions: { timeout: consts.FRAMEWORK_HANDLER_TIMEOUT }});
```
Comment (eladb) [inline packages/@aws-cdk/custom-resources/lib/provider-framework/runtime/outbound.ts:28]: add a comment explaining why this is needed

File: packages/@aws-cdk/custom-resources/lib/provider-framework/runtime/outbound.ts [hunk 2]
```diff
@@ -32,6 +35,8 @@ async function defaultInvokeFunction(req: AWS.Lambda.InvocationRequest): Promise
     lambda = new AWS.Lambda();
   }
 
+  lambda.config.update({httpOptions: { timeout: consts.FRAMEWORK_HANDLER_TIMEOUT }});
```
Comment (eladb) [inline packages/@aws-cdk/custom-resources/lib/provider-framework/runtime/outbound.ts:38]: same

Comment (eladb) [reply packages/@aws-cdk/custom-resources/lib/provider-framework/runtime/outbound.ts:38]: Can we do it in `new AWS.Lambda` instead?

File: packages/@aws-cdk/custom-resources/lib/provider-framework/runtime/consts.ts [hunk 3]
```diff
@@ -5,3 +5,4 @@ export const WAITER_STATE_MACHINE_ARN_ENV = 'WAITER_STATE_MACHINE_ARN';
 export const FRAMEWORK_ON_EVENT_HANDLER_NAME = 'onEvent';
 export const FRAMEWORK_IS_COMPLETE_HANDLER_NAME = 'isComplete';
 export const FRAMEWORK_ON_TIMEOUT_HANDLER_NAME = 'onTimeout';
+export const FRAMEWORK_HANDLER_TIMEOUT = 900000; // 15 minutes
```
Comment (eladb) [inline packages/@aws-cdk/custom-resources/lib/provider-framework/runtime/consts.ts:8]: Why is this needed here? Feels like it could be internal to `outbound.ts`? This file is the "protocol" between the construct and the handler.

Comment (eduardomourar) [reply packages/@aws-cdk/custom-resources/lib/provider-framework/runtime/consts.ts:8]: no real reason, except to keep the constants in a single place

Comment (eladb) [reply packages/@aws-cdk/custom-resources/lib/provider-framework/runtime/consts.ts:8]: The file name is misleading. Please move locally.

Comment (eladb) [reply packages/@aws-cdk/custom-resources/lib/provider-framework/runtime/consts.ts:8]: Still waiting for this

**Comment by eladb** [comment]
Build is still failing
